### PR TITLE
Properly display falsy `concurrency_limit` value in CLI

### LINKED ISSUE #12355: CLI incorrectly shows work-queue concurrency limi as None when set to 0
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

The concurrency limit of `my-queue` is set to `0` in the UI.

```
❯ prefect -p prd work-queue ls --pool eks-europe-1
              Work Queues in Work Pool 'eks-europe-1'               
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
┃ Name                              ┃ Priority ┃ Concurrency Limit ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
│ my-queue                          │ 6        │ None              │
│ default                           │ 1        │ 100               │
└───────────────────────────────────┴──────────┴───────────────────┘
```

### Reproduction

```python3
N/A
```

### Error

_No response_

### Versions

```Text
server = 2.16.4
client = 2.16.4


❯ prefect version
Version:             2.16.4
API version:         0.8.4
Python version:      3.10.10
Git commit:          e3e7df9d
Built:               Thu, Mar 14, 2024 5:11 PM
OS/Arch:             darwin/x86_64
Profile:             default
Server type:         ephemeral
Server:
  Database:          sqlite
  SQLite version:    3.39.4

```
```


### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hainenber**
Fixes #12355 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: tests/cli/test_work_queues.py [hunk 1]
```diff
@@ -579,6 +579,20 @@ def test_ls_with_pool(
...
+            work_queue_1,
+    ):
+        invoke_and_assert(
+            command=f"work-queue set-concurrency-limit {work_queue_1.name} 0",
+            expected_code=0,
+        )
+        invoke_and_assert(
+            command=f"work-queue ls -p {work_queue_1.work_pool.name}",
+            expected_code=0,
+            expected_output_does_not_contain='None',
```
Comment (zzstoatzz) [inline tests/cli/test_work_queues.py:593]: could we add a check here that the output _does_ contain 0 instead of `None`?

**Review by zzstoatzz — CHANGES_REQUESTED** [review]
nice catch! thank you - just one request

File: tests/cli/test_work_queues.py (hunk 1 above)
Comment (zzstoatzz) [reply tests/cli/test_work_queues.py:593]: actually - nevermind, because there will be a zero elsewhere which makes the check not worth anything

Comment (hainenber) [reply tests/cli/test_work_queues.py:593]: Sure can do!

My little concern is that there might be other places in the `ls` output having 0

**Comment by zzstoatzz** [comment]
thanks again @hainenber - can you go ahead and `pre-commit install` and `pre-commit run --all-files` so we pass static analysis?

**Comment by hainenber** [comment]
Done :D

**Comment by zzstoatzz** [comment]
hey @hainenber - it looks like this test is failing (I assume for a similar reason to what we mentioned earlier, i.e. `None` is present elsewhere)
> FAILED tests/cli/test_work_queues.py::TestLS::test_ls_with_zero_concurrency_limit - AssertionError: Undesired contents 'None' found in CLI output

let me know if you have trouble updating that test

**Comment by hainenber** [comment]
Hi, I've fixed the flaky test in latest commit. PTAL when you have time. Thanks!

**Review by zzstoatzz — APPROVED** [review]
thanks so much @hainenber for the contribution and the quick revisions! 💙

**Pull request merged**